File: 프로그래머스/bjw/타겟 넘버.js
let len = 0;
let answer = 0;
const dfs = (count, num, numbers, target) => {
  if (count === len) {
    num === target ? answer++ : null;
    return;
  }
  count++;
  const arr = [numbers[count], -numbers[count]];
  arr.forEach((next) => dfs(count, num + next, numbers, target));
};

function solution(numbers, target) {
  len = numbers.length - 1;
  dfs(0, numbers[0], numbers, target);
  dfs(0, -numbers[0], numbers, target);
  return answer;
}

File: 프로그래머스/jsh/python/타겟넘버.py
total = 0
def dfs(numbers, target, sum_, n):
    global total
    if n == len(numbers):
        if sum_ == target:
            total += 1
        return
    
    dfs(numbers, target, sum_ + numbers[n], n+1)
    dfs(numbers, target, sum_ - numbers[n], n+1)
    
def solution(numbers, target):
    answer = 0
    # 타켓 넘버 만들기
    dfs(numbers, target, numbers[0], 1) # 더하거나
    dfs(numbers, target, -numbers[0], 1) # 빼서
    answer = total
    return answer